Document SHOW FUNCTIONS and SHOW PROCEDURES (#1089)
neo4j/neo4j-documentation: cypher/cypher-docs/src/docs/dev/ql/listing/listing-procedures-columns.adoc

.List procedures output
[options="header", width="100%", cols="1a,4,^.^,^"]
|===
| Column
| Description
| Default output
| Full output

| `name`
| The name of the procedure.
| `+`
| `+`

| `description`
| The procedure description.
| `+`
| `+`

| `mode`

> | The procedure mode, for example `READ` or `WRITE`.

Reviewer: Is there a reference for procedure modes we can link to from here?

Author: maybe, but probably not, can see if I can find anything

Author: They are described in the java reference in the section about writing [user-defined procedures](https://neo4j.com/docs/java-reference/current/extending-neo4j/procedures-and-functions/procedures/#writing-procedure), but I don't find anything else when searching the docs

| `+`
| `+`

| `worksOnSystem`
| Whether the procedure can be run on the `system` database or not.
| `+`
| `+`

| `signature`
| The signature of the procedure.
|
| `+`

| `argumentDescription`
| List of the arguments for the procedure, as map of strings with name, type, default, and description.
|
| `+`

| `returnDescription`
| List of the returned values for the procedure, as map of strings with name, type, and description.
|
| `+`

| `admin`

> | If it is or isn't an admin procedure.

Reviewer: ```suggestion
`true` if this procedure is an admin procedure
```

|
| `+`

| `rolesExecution`

> | List of roles who can execute this procedure.

Reviewer: ```suggestion
| List of roles permitted to execute this procedure.
```

Will be `null` without the <<administration-security-administration-dbms-privileges-role-management, SHOW ROLE>> privilege.
|
| `+`

| `rolesBoostedExecution`

> | List of roles who will use boosted mode when executing this procedure.

Reviewer: ```suggestion
| List of roles permitted to use boosted mode when executing this procedure.
```
